zebra/zebra_srv6_vty.c

```diff
@@ -0,0 +1,342 @@
+	json_object *json_locator = NULL;
+
+	if (uj) {
+		json = json_object_new_object();
+		json_locators = json_object_new_array();
+		json_object_object_add(json, "locators", json_locators);
+
+		for (ALL_LIST_ELEMENTS_RO(srv6->locators, node, locator)) {
+			prefix2str(&locator->prefix, str, sizeof(str));
+			json_locator = json_object_new_object();
+			json_object_string_add(json_locator, "name",
+					       locator->name);
+			json_object_string_add(json_locator, "prefix", str);
+			json_object_array_add(json_locators, json_locator);
+		}
+
+		vty_out(vty, "%s\n", json_object_to_json_string_ext(json,
+					JSON_C_TO_STRING_PRETTY));
+		json_object_free(json);
+	} else {
+		vty_out(vty, "Locator:\n");
+		vty_out(vty, "Name                 ID      Prefix                   Status\n");
+		vty_out(vty, "-------------------- ------- ------------------------ -------\n");
+
+		id = 1;
+		for (ALL_LIST_ELEMENTS_RO(srv6->locators, node, locator)) {
+			prefix2str(&locator->prefix, str, sizeof(str));
+			vty_out(vty, "%-20s %7d %-24s Up\n",
+				locator->name, id, str);
+			++id;
+		}
+		vty_out(vty, "\n");
+	}
+
+	return CMD_SUCCESS;
+}
+
+DEFUN_NOSH (segment_routing,
+            segment_routing_cmd,
+            "segment-routing",
+            "Segment Routing\n")
+{
+	vty->node = SR_NODE;
+	return CMD_SUCCESS;
+}
+
+DEFUN_NOSH (srv6,
+            srv6_cmd,
+            "srv6",
+            "Segment Routing SRv6\n")
+{
+	vty->node = SRV6_NODE;
+	return CMD_SUCCESS;
+}
+
+DEFUN_NOSH (srv6_locators,
+            srv6_locators_cmd,
+            "locators",
+            "Segment Routing SRv6 locators\n")
+{
+	vty->node = SRV6_LOCS_NODE;
+	return CMD_SUCCESS;
+}
+
+DEFUN_NOSH (srv6_locator,
+            srv6_locator_cmd,
+            "locator WORD",
+            "Segment Routing SRv6 locator\n"
+            "Specify locator-name\n")
+{
+	struct srv6_locator *locator = NULL;
+
+	locator = zebra_srv6_locator_lookup(argv[1]->arg);
+	if (locator) {
+		VTY_PUSH_CONTEXT(SRV6_LOC_NODE, locator);
+		return CMD_SUCCESS;
+	}
+
+	locator = srv6_locator_alloc(argv[1]->arg);
+	if (!locator) {
+		vty_out(vty, "%% Alloc failed\n");
+		return CMD_WARNING_CONFIG_FAILED;
+	}
+
+	VTY_PUSH_CONTEXT(SRV6_LOC_NODE, locator);
+	vty->node = SRV6_LOC_NODE;
+	return CMD_SUCCESS;
+}
+
+DEFUN (locator_prefix,
+       locator_prefix_cmd,
+       "prefix X:X::X:X/M [func-bits (8-64)]",
+       "Configure SRv6 locator prefix\n"
+       "Specify SRv6 locator prefix\n"
+       "Configure SRv6 locator function length in bits\n"
+       "Specify SRv6 locator function length in bits\n")
+{
+	VTY_DECLVAR_CONTEXT(srv6_locator, locator);
+	struct prefix_ipv6 prefix;
+	uint8_t function_bits_length = 16;
+	int ret;
+
+	ret = str2prefix_ipv6(argv[1]->arg, &prefix);
+	if (ret <= 0) {
+		vty_out(vty, "%% Malformed address\n");
+		return CMD_WARNING_CONFIG_FAILED;
+	}
+	apply_mask_ipv6(&prefix);
+
+	if (argc >= 3)
+		function_bits_length = strtoul(argv[3]->arg, NULL, 10);
```

Comment:
Not a blocking comment but you really ought to use defpys here

Reply:
I see. I will re-write this function with DEFPY.

Reply:
Re-wrote in this commit (https://github.com/slankdev/frr/commit/ad75b3eb367e66b0f35aca7f7adde8251df15865)